igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/BuildController.groovy

```diff
@@ -92,13 +94,16 @@ class BuildController {
     }
 
     @RequestMapping(value = '/builds/all/{master}/**')
-    List<Build> getBuilds(@PathVariable String master, HttpServletRequest request) {
-        def job = (String) request.getAttribute(
+    List<Object> getBuilds(@PathVariable String master, HttpServletRequest request) {
+      def job = (String) request.getAttribute(
             HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE).split('/').drop(4).join('/')
-        if (!masters.map.containsKey(master)) {
-            throw new MasterNotFoundException("Master '${master}' not found")
-        }
-        masters.map[master].getBuilds(job).list
+      if (buildMasters.filteredMap(BuildServiceProvider.JENKINS).containsKey(master)) {
+        buildMasters.map[master].getBuilds(job).list
+        } else if (buildMasters.filteredMap(BuildServiceProvider.TRAVIS).containsKey(master)) {
```

Comment:
What it someone has a jenkins and travis master with the same name? It seems like they will only be able to interact with the jenkins master.

Maybe we can specify which CI system to talk to as an optional query parameter, where jenkins is assumed by default if none is specified. That way you don't break existing contracts between the other services and igor, but you can correctly designate the buildserviceprovider when needed.